src/betterproto/plugin/parser.py

```diff
@@ -0,0 +1,180 @@
+            # Create a new output if there is no output for this package
+            request_data.output_packages[output_package_name] = OutputTemplate(
+                parent_request=request_data, package_proto_obj=proto_file
+            )
+        # Add this input file to the output corresponding to this package
+        request_data.output_packages[output_package_name].input_files.append(proto_file)
+
+    # Read Messages and Enums
+    # We need to read Messages before Services in so that we can
+    # get the references to input/output messages for each service
+    for output_package_name, output_package in request_data.output_packages.items():
+        for proto_input_file in output_package.input_files:
+            for item, path in traverse(proto_input_file):
+                read_protobuf_type(item=item, path=path, output_package=output_package)
+
+    # Read Services
+    for output_package_name, output_package in request_data.output_packages.items():
+        for proto_input_file in output_package.input_files:
+            for index, service in enumerate(proto_input_file.service):
+                read_protobuf_service(service, index, output_package)
+
+    # Generate output files
+    output_paths: pathlib.Path = set()
+    for output_package_name, template_data in request_data.output_packages.items():
+
+        # Add files to the response object
+        output_path = pathlib.Path(*output_package_name.split("."), "__init__.py")
+        output_paths.add(output_path)
+
+        f: response.File = response.file.add()
+        f.name: str = str(output_path)
+
+        # Render and then format the output file
+        f.content: str = black.format_str(
+            template.render(description=template_data),
+            mode=black.FileMode(target_versions={black.TargetVersion.PY37}),
+        )
+
+    # Make each output directory a package with __init__ file
+    init_files = (
+        set(
+            directory.joinpath("__init__.py")
+            for path in output_paths
+            for directory in path.parents
+        )
+        - output_paths
+    )
+
+    for init_file in init_files:
+        init = response.file.add()
+        init.name = str(init_file)
+
+    for output_package_name in sorted(output_paths.union(init_files)):
+        print(f"Writing {output_package_name}", file=sys.stderr)
+
+
+def read_protobuf_type(
+    item: DescriptorProto, path: List[int], output_package: OutputTemplate
+) -> None:
+    if isinstance(item, DescriptorProto):
+        if item.options.map_entry:
+            # Skip generated map entry messages since we just use dicts
+            return
+        # Process Message
+        message_data = Message(parent=output_package, proto_obj=item, path=path)
+        for index, field in enumerate(item.field):
+            if is_map(field, item):
+                MapField(parent=message_data, proto_obj=field, path=path + [2, index])
```

Comment:
Yeah either compiler or Parsed..Definition sound good to me.

Reply:
With "Compiler" it would be:
- RequestCompiler
- MessageCompiler
- FieldCompiler
- MapEntryCompiler
- OneOfFieldCompiler
- EnumDefCompiler
- ServiceCompiler
- ServiceMethodCompiler

Right?

Reply:
> With "Compiler" it would be:
> - RequestCompiler
> - MessageCompiler
> - FieldCompiler
> - MapEntryCompiler
> - OneOfFieldCompiler
> - EnumDefCompiler
> - ServiceCompiler
> - ServiceMethodCompiler
> 
> Right?

Yes :+1:

Although I haven't looked at protobuf descriptions to see if the classnames match, but if it doesn't, it would be an easy fix. So this looks fine to me.

EnumDefCompiler could just be EnumCompiler, I guess?

Reply:
The reason I suggested EnumDefCompiler instead of EnumCompiler is that _I_ originally was confused because I thought the enum objects protoc was supplying were enum fields or something like that instead of the actual enum defenitions. But maybe that was just me not knowing enough about protobuf structures... Again I really am open to anything you or others decide on 😃

Reply:
Ah right, I understand. I may have had the same confusing, until I realized enums are also independent objects.

What further confuses things is that Enum values are also distinct from the Enum definition itself, although I think the Enum values end up being just fields on the Enum, but not sure (?).

How about EnumTypeCompiler?

Reply:
Yep "Enum Fields" are just Fields of the type <EnumDef name> if that makes sense. They're treated just like any other field. So the only time you encounter anything "Enum" related is the enum definitions. Why not `EnumDefinitionCompiler`? It's a bit long but I think it's most explicit. But otherwise `EnumTypeCompiler` sounds fine as well.